### DISCUSSION TITLE
'use client' Wrapper with Children?

### DISCUSSION THREAD

**Discussion by michael-mcgreal** [vercel/next.js#51379] [App Router]
To avoid creating a new client component file for every `<input />`, I am attempting to create a 'use client' **Wrapper component that accepts children**. Is this not possible?

page.tsx
```ts
export default async function Page() {
  // some server code
  return (
    <div>
      <ClientWithChildren>
        <input type='text' onChange={() => console.log('client code')} />
      </ClientWithChildren>
    </div>
  )
}
```

ClientWithChildren.tsx
```ts
'use client'
export default function ClientWithChildren({
  children,
}: {
  children: React.ReactNode
}) {
  return <>children</>
}
```

**Comment by icyJoseph** [comment]
Nope, because the input element is imported into `Page`, not into `ClientWithChildren`. The Server/Client boundary is at the module level.
